librz/core/cmd/cmd_analysis.c

```diff
@@ -1778,21 +1778,21 @@ RZ_IPI RzCmdStatus rz_analysis_syscall_show_handler(RzCore *core, int argc, cons
 	return RZ_CMD_STATUS_OK;
 }
 
-static const char *syscallNumber(int n) {
-	return sdb_fmt(n > 1000 ? "0x%x" : "%d", n);
-}
+#define SYSCALL_NUM_FMT(n) ((n) > 1000 ? "0x%x" : "%d")
 
 RZ_IPI RzCmdStatus rz_analysis_syscall_print_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	char tmpbuf[32] = { 0 };
 	RzListIter *iter;
 	RzSyscallItem *si;
 	RzList *list = rz_syscall_list(core->analysis->syscall);
 	rz_cmd_state_output_array_start(state);
 	rz_list_foreach (list, iter, si) {
 		switch (state->mode) {
-		case RZ_OUTPUT_MODE_STANDARD:
-			rz_cons_printf("%s = 0x%02x.%s\n",
-				si->name, si->swi, syscallNumber(si->num));
+		case RZ_OUTPUT_MODE_STANDARD: {
+			snprintf(tmpbuf, sizeof(tmpbuf)), SYSCALL_NUM_FMT(si->num), si->num); 
+			rz_cons_printf("%s = 0x%02x.%s\n", si->name, si->swi, tmpbuf);
```

Comment:
instead of doing this, why not just replace that with this?
```suggestion
			snprintf(tmpbuf, sizeof(tmpbuf)), SYSCALL_NUM_FMT(si->num), si->num); 
			if (n > 1000) {
				rz_cons_printf("%s = 0x%02x.0x%x\n", si->name, si->swi, si->num);
			} else {
				rz_cons_printf("%s = 0x%02x.%d\n", si->name, si->swi, si->num);
			}
```

Reply:
The pattern repeats in many places. The old `sdb_fmt()`-based solution was much more convenient in this particular case...

Reply:
i would strongly implement this everywhere. at the end of the day is better than a call to a function to get the formatting.